adbc.h

```diff
@@ -479,7 +485,7 @@ struct ADBC_EXPORT AdbcDatabase {
   void* private_data;
   /// \brief The associated driver (used by the driver manager to help
   ///   track state).
-  struct AdbcDriver* private_driver;
+  void* private_driver;
```

Comment:
isn't this technically a breaking change?

Reply:
Yes; I think we could avoid this if we wanted, but it would require acrobatics

Reply:
(basically: have the new function table be embedded as the private_data of the AdbcDriver)

Reply:
Are there any other libraries that you're aware of that have done something like this without an embedding 'treadmill' that we could use for inspiration?

Reply:
UCX has you pass in sizeof(struct) everywhere, basically, so it knows what fields are safe to access or not.

Reply:
And then ODBC, basically, assumes you will use the driver manager. I'm not familiar with the driver implementation side but from what I see there's no use of a table like this (which would have placed different constraints on us).